### DISCUSSION TITLE
Object-based permissions based on Tenant Group

### DISCUSSION THREAD

**Discussion by billfehring** [netbox-community/netbox#5483] [Q&A] [answered]
Hi everyone,

I'm trying to use the Tenant Group to assign permissions to objects belonging to any tenant within a tenant group. The Tenant Group does not appear to be directly associated with the object (in this case, a Rack) nor do the current examples in the documentation seem to describe how I might go about this. 

I think what I need to do is somehow recursively get a flattened array of all tenants that exist within a tenant group (which according to the documentation can also be hierarchical) and do something like this:

`{"tenant__in": tentants.lab_tenants}`

But for someone not familiar with the Django ORM, I'm having a bit of trouble. Any ideas?

Thanks!

Bill Fehring

**Comment by jeremystretch** [comment, marked as answer]
You can access related objects using a dunder (double underscore). So for example, to match all objects assigned to a tenant within the tenant group "Customers", you could do `{"tenant__group__name": "Customers"}`.

**Reply by billfehring** [reply]
Thank you @jeremystretch that was exactly what I needed!

Happy Holidays!

**Reply by proudbro** [reply]
... or do it this way `{"tenant__group__id": 1}` 
You can get your Tenant Group ID's by the endpoint ` https://netbox_name/api/tenancy/tenant-groups/` 

It should prevent you from being broken permission due to renaming Tenant Group
